Collector/MigrationsCollector.php

```diff
@@ -0,0 +1,170 @@
+ *
+ * (c) Michael Simonson <mike@simonson.be>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Doctrine\Bundle\MigrationsBundle\Collector;
+
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Migrations\Configuration\Configuration;
+use Doctrine\DBAL\Migrations\Tools\Console\Helper\MigrationStatusInfosHelper;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollectorInterface;
+
+class MigrationsCollector implements DataCollectorInterface
+{
+    /** @var  string */
+    private $migrationTablename;
+
+    /** @var  string */
+    private $migrationName;
+
+    /** @var Connection  */
+    private $connection;
+
+    /** @var  string */
+    private $migrationNamespace;
+
+    /** @var  string */
+    private $migrationDirectory;
+
+    public function __construct(Connection $connection, $migrationNamespace, $migrationDirectory, $migrationTablename, $migrationName)
+    {
+        $this->connection = $connection;
+        $this->migrationNamespace = $migrationNamespace;
+        $this->migrationDirectory = $migrationDirectory;
+        $this->migrationTablename = $migrationTablename;
+        $this->migrationName = $migrationName;
+    }
+
+    /**
+     * Collects data for the given Request and Response.
+     *
+     * @param Request $request A Request instance
+     * @param Response $response A Response instance
+     * @param \Exception $exception An Exception instance
+     */
+    public function collect(Request $request, Response $response, \Exception $exception = null)
+    {
+        $configuration = new Configuration($this->connection);
+        $configuration->setMigrationsNamespace($this->migrationNamespace);
+        $configuration->setMigrationsDirectory($this->migrationDirectory);
+        $configuration->setMigrationsTableName($this->migrationTablename);
+        $configuration->setName($this->migrationName);
+        $configuration->registerMigrationsFromDirectory($configuration->getMigrationsDirectory());
+        $migrationsStatusInfos = new MigrationStatusInfosHelper($configuration);
+        $this->data = $migrationsStatusInfos->getMigrationsInfos();
+        $newMigrationsList = $configuration->getMigrationsToExecute('up', $configuration->getLatestVersion());
+        $this->data['newMigrationsList'] = array_map(function($migration) {
+            return $migration->getVersion();
+        }, $newMigrationsList);
+        $this->data['executedUnavailableMigrationsList'] = array_map(function($migration) {
+            return $migration->getVersion();
+        }, $migrationsStatusInfos->getExecutedUnavailableMigrations());
+
+        $this->connection = null;
+        $this->migrationDirectory = null;
+        $this->migrationNamespace = null;
+        $this->migrationName = null;
+        $this->migrationTablename = null;
```

Comment:
I think resetting them here can cause issues when subrequests are used (as they also collect the info)